bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/handler/FreeAtHomeBridgeHandler.java

```diff
@@ -0,0 +1,819 @@
+        try {
+            Request req = httpClient.newRequest(url);
+
+            req.content(new StringContentProvider(valueString));
+            req.method(HttpMethod.PUT);
+
+            logger.debug("Set datapoint by url: {} value: {}", url, valueString);
+
+            ContentResponse response = req.send();
+
+            logger.debug("Communication result [{}]", response.getStatus());
+
+            if (response.getStatus() != 200) {
+                logger.error("Communication error by setDatapoint [{}]", response.getStatus());
+
+                restartHttpConnection();
+            }
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            logger.error("Communication error by setDatapoint [{}]", e.getMessage());
+
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+
+        return true;
+    }
+
+    /**
+     * Method to process socket events
+     *
+     * @author Andras Uhrin
+     *
+     */
+    @SuppressWarnings("deprecation")
+    public void processSocketEvent(String receivedText) {
+        JsonReader reader = new JsonReader(new StringReader(receivedText));
+
+        reader.setLenient(true);
+
+        JsonParser parser = new JsonParser();
+        JsonElement jsonTree = parser.parse(reader);
+
+        // check the output
+        if (jsonTree.isJsonObject()) {
+            JsonObject jsonObject = jsonTree.getAsJsonObject();
+
+            jsonObject = jsonObject.getAsJsonObject(sysApUID);
+            jsonObject = jsonObject.getAsJsonObject("datapoints");
+
+            Set<String> keys = jsonObject.keySet();
+
+            Iterator<String> iter = keys.iterator();
+
+            while (iter.hasNext()) {
+                String eventDatapointID = iter.next();
+
+                JsonElement element = jsonObject.get(eventDatapointID);
+                String value = element.getAsString();
+
+                channelUpdateHandler.updateChannelByDatapointEvent(eventDatapointID, value);
+
+                logger.debug("Socket event processed: event-datapoint-ID {} value {}", eventDatapointID, value);
+            }
+        }
+    }
+
+    /**
+     * Method to open Http connection
+     *
+     * @author Andras Uhrin
+     *
+     */
+    public boolean openHttpConnection() {
+        boolean ret = false;
+
+        // Configure client
+        httpClient.setFollowRedirects(false);
+        httpClient.setMaxConnectionsPerDestination(1);
+        httpClient.setMaxRequestsQueuedPerDestination(50);
+
+        // Set timeouts
+        httpClient.setIdleTimeout(-1);
+        httpClient.setConnectTimeout(5000);
+
+        try {
+            // Start HttpClient.
+            switch (httpClient.getState()) {
+                case AbstractLifeCycle.FAILED:
+                case AbstractLifeCycle.STOPPED: {
+                    httpClient.start();
```

Comment:
See comment in `FreeAtHomeSystemHandlerFactory`. Consider letting the factory manage life cycle.